### PR TITLE
Convert date and time values to DateTimeImmutable if possible, dependency updates

### LINKED ISSUE #52: Return date values as DateTimeImmutable
Date and time values returned should not be subject to change.

### PR DESCRIPTION
**Pull request by szymach**
Resolves #52

### CODE REVIEW

**Comment by szymach** [comment]
@fsi-open

**Comment by szymach** [comment]
@fsi-open

**Comment by szymach** [comment]
@fsi-open

File: Model/Resource.php [hunk 1]
```diff
@@ -199,4 +202,13 @@ public function getBoolValue()
     {
         return $this->boolValue;
     }
+
+    /**
+     * @param DateTime $value
+     * @return DateTimeImmutable
+     */
+    private function convertDateToImmutable(DateTime $value)
```
Comment (chives) [inline Model/Resource.php]: I would use `DateTimeInterface` as typehint because it would be compatible with doctrine's immutable types

**Comment by szymach** [comment]
merge?

**Pull request merged**